mqtt/mqtt-broker/src/configuration/size.rs

```diff
@@ -0,0 +1,187 @@
+use std::str::FromStr;
+
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Deserializer};
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct HumanSize(usize);
+
+impl HumanSize {
+    pub const fn new(bytes: usize) -> Self {
+        Self(bytes)
+    }
+
+    pub fn new_kilobytes(kilobytes: usize) -> Option<Self> {
+        kilobytes.checked_mul(1024).map(Self)
```

Comment:
nit: I understand your urge to use combinators everywhere, but isn't it simpler to read and understand `Self( kilobytes.checked_mul(1024))`?

Reply:
nvm, you still need to handle Option...